pkg/server/backend_manager.go

```diff
@@ -121,17 +120,39 @@ func (b *backend) Context() context.Context {
 	return b.conn.Context()
 }
 
-func newBackend(conn agent.AgentService_ConnectServer) *backend {
+func (b *backend) GetAgentIdentifiers() (header.Identifiers, error) {
+	var agentIdentifiers header.Identifiers
+	md, ok := metadata.FromIncomingContext(b.Context())
+	if !ok {
+		return agentIdentifiers, fmt.Errorf("failed to get context")
```

Comment:
nit: "failed to get metadata" might be more accurate?
```suggestion
		return agentIdentifiers, fmt.Errorf("failed to get metadata from context")
```

Reply:
Done.